src/chain_sync/tipset_syncer.rs

```diff
@@ -844,73 +843,75 @@ async fn sync_headers_in_reverse<DB: Blockstore + Sync + Send + 'static>(
             .await
             .map_err(TipsetRangeSyncerError::NetworkTipsetQueryFailed)?;
 
-        for tipset in network_tipsets {
-            // Break if have already traversed the entire tipset range
-            if tipset.epoch() < current_head.epoch() {
-                break 'sync;
-            }
-            validate_tipset_against_cache(bad_block_cache, tipset.key(), &parent_blocks)?;
-            parent_blocks.extend(tipset.cids());
+        for tipset in network_tipsets
+            .into_iter()
+            .take_while(|ts| ts.epoch() >= until_epoch)
+        {
+            validate_tipset_against_cache(bad_block_cache, tipset.key(), &accepted_blocks)?;
+            accepted_blocks.extend(tipset.cids());
             tracker.write().set_epoch(tipset.epoch());
             parent_tipsets.push(tipset);
         }
     }
     drop(wp);
 
-    let oldest_tipset = parent_tipsets.last().clone();
-    // Determine if the local chain was a fork.
-    // If it was, then sync the fork tipset range by iteratively walking back
+    let oldest_tipset = parent_tipsets.last();
```

Comment:
nit: perhaps `ordest_parent` is a better name, similar to https://github.com/ChainSafe/forest/pull/4291/files#diff-c2e102581e8811dd894188425099a003280f0c7e38b9c5a7ca75f2a0dd0bdaddR824

Reply:
Good point, actually `parent` is quite confusing here, renamed `parent_tipsets` to `pending_tipsets(_for_sync)`, and `oldest_tipset` to `oldest_pending_tipset`